lupa/_lupa.pyx

```diff
@@ -819,6 +829,41 @@ cdef class _LuaTable(_LuaObject):
             lua.lua_settop(L, old_top)
             unlock_runtime(self._runtime)
 
+    @cython.final
+    cdef tuple _unpack(self):
+        """Unpacks table into a tuple of positional arguments
+        and a dictionary of keyword arguments
+        """
+        assert self._runtime is not None
+        cdef tuple args
+        cdef dict kwargs = {}
+        cdef bytes encoding
+        cdef Py_ssize_t index, length
+        lock_runtime(self._runtime)
+        try:
+            encoding = self._runtime._source_encoding
+            length = <Py_ssize_t>self._len()
+            args = cpython.tuple.PyTuple_New(length)
+            for key, value in self.items():
+                if isinstance(key, (int, long)) and not isinstance(key, bool):
+                    index = <Py_ssize_t>key
+                    if index < 1 or index > length:
+                        raise IndexError("table index out of range")
+                    cpython.ref.Py_INCREF(value)
+                    cpython.tuple.PyTuple_SET_ITEM(args, index-1, value)
+                elif isinstance(key, bytes):
+                    if IS_PY2:
+                        kwargs[key] = value
+                    else:
+                        kwargs[(<bytes>key).decode(encoding)] = value
```

Comment:
BTW, I think this is (and was) buggy. `key` can only be `bytes` here if the source encoding is None (otherwise it already got decoded in `py_from_lua()`), and in that case, `.decode()` will reject the invalid argument. We can probably just ignore the `bytes` case. But `py_from_lua()` still needs the encoding.

Reply:
But `self._source_encoding is not None` because (on `LuaRuntime.__cinit__`)
```cython
self._source_encoding = _asciiOrNone(source_encoding) or self._encoding or b'UTF-8'
```